aten/src/ATen/native/cuda/ForeachFunctors.cuh

```diff
@@ -179,9 +180,9 @@ __device__ __forceinline__ void pointwise_op_scalar(
 //
 // Binary Functors
 //
-template<typename T, int depth, int r_args_depth, int res_arg_index>
+template<typename T, int depth, int r_args_depth, int res_arg_index, bool disable_opmath_type=false>
```

Comment:
What is this change for?

Reply:
I saw numerical differences of the outputs from `multi_tensor_apply` and the native for fp16 and bf16 inputs (`opmath_t` is float32 for float16 and bfloat16, scalar_type for the other scalar_type's) so I concluded power kernels don't upcast low-precision numbers.
Is my understanding correct?

Reply:
I'm not sure to understand? You don't want these to run with high precision intermediary? Why?

Reply:
To implement the same behavior as the native kernels.
e.g. https://github.com/pytorch/pytorch/blob/5b2afaaca8ca36c3029e22664dd63ee1cf2461b4/aten/src/ATen/native/cuda/PowKernel.cu#L124-L141 doesn't seem to use float32 for float16 and bfloat16 inputs.

Reply:
cc @ngimel is this a bug or a feature? :)

Reply:
From offline chat:
- There shouldn't be any measurable numerical difference between the two for a single op.
- Is there any measurable perf difference between upcasting or not? If so, let's go with the fastests.
- If not, we should just go with the simplest implementation.

Reply:
from offline discussion, pow_scalar_tensor_impl call into pow_ which does convert to float before calling std::pow.
So we should be doing the same here in the end.